### ISSUE TITLE
Sound too fast for me too

### ISSUE THREAD

**Issue by BugHunterPhilosopher** [badaix/snapcast#631]
Hi there,

I'm sorry to bug you with this one but I have looked at all the other issues regarding sound too fast and nothing there helps. Here is my problem: the sound on my client is way too fast, maybe twice too much.

I have a MPD client and a Clementine Player on the server and the clients are on the same (Debian Linux 10) machine for the first one and a Raspberry Pi 3 running Retropie (Raspbian-based distro) for the second one. Both clients exhibit the same behaviour.

I have set up the sample rates to 44100Hz everywhere but it doesn't help. I use Pulseaudio, too.

You'll find attached my /etc/snapserver.conf, .asoundrc, /etc/default/snapserver and /etc/default/snapclient.

[conf.zip](https://github.com/badaix/snapcast/files/4708057/conf.zip)

Thanks for any help.

**Comment by badaix** [comment]
Twice as fast sounds like you're feeding a mono signal in

**Comment by BugHunterPhilosopher** [comment]
Strangely, with 22050Hz, it still sounds twice too fast. The good value in terms of speed sounds to be 8268Hz. But the sound is horrible. In Clementine, with a frequency of 192KHz the sound is a little better than with auto or 44100, though.

Any idea?

**Comment by BugHunterPhilosopher** [comment]
Thanks for being so fast, by the way! :-)

**Comment by badaix** [comment]
I have no idea about asound configs, this is discussed here from time to time and mentioned [here](https://github.com/badaix/snapcast/blob/master/doc/player_setup.md#alsa), but yours looks weird to me (my comments inside `<--`):

```
defaults.ctl.card 3
defaults.pcm.card 3
defaults.timer.card 3

pcm.!default {
	type plug
	slave.pcm rate44100Hz <-- I think this is kind of a link to the non-existing "pcm.rate44100Hz" config
}

pcm.rate48000Hz {  <-- here you have 48000Hz (it's just a name, I guess)
	type rate
	slave {
		pcm writeFile # Direct to the plugin which will write to a file
		format S16_LE
		rate 441000 <-- this is wrong, 44100 should be better
	}
}

pcm.writeFile {
	type file
	slave.pcm null
	file "/tmp/snapfifo"
	format "raw"
}
```

**Comment by BugHunterPhilosopher** [comment]
Good catch! Unfortunately it is not enough... Here is my new conf:

defaults.ctl.card 3
defaults.pcm.card 3
defaults.timer.card 3

pcm.!default {
	type plug
	slave.pcm rate44800Hz
}

pcm.rate44800Hz {
	type rate
	slave {
		pcm writeFile # Direct to the plugin which will write to a file
		format S16_LE
		rate 44800
	}
}

pcm.writeFile {
	type file
	slave.pcm null
	file "/tmp/snapfifo"
	format "raw"
}


The problem remains the same. Any more ideas, please?

**Comment by badaix** [comment]
please use 44100 or 48000, but not 44800, this is not standard and might not be supported by your hw or audio stack.

**Comment by BugHunterPhilosopher** [comment]
Sorry about that, it was late. That's fixed with 48000, but still the same.

Just an idea: shall I set up a latency time?

**Comment by badaix** [comment]
Changing latency shouldn't help

**Comment by BugHunterPhilosopher** [comment]
I don't get a clue, then. :-(

**Comment by badaix** [comment]
Maybe you should add `channels 2`

**Comment by BugHunterPhilosopher** [comment]
I he this now:

pcm.!default {
	type plug
	slave.pcm rate44800Hz
}

pcm.rate44800Hz {
	type rate
	slave {
		pcm writeFile # Direct to the plugin which will write to a file
		format S16_LE
		rate 48000
		channels 2
	}
}

pcm.writeFile {
	type file
	slave.pcm null
	file "/tmp/snapfifo"
	format "raw"
}

Still no way... Maybe my 5.1 setup can be the culprit in a way or another?

**Comment by badaix** [comment]
For whatever reason this is not working, don't use the number 44800 in your config.

**Comment by BugHunterPhilosopher** [comment]
Yes, sorry about that one. I have this, now:

defaults.ctl.card 3
defaults.pcm.card 3
defaults.timer.card 3

pcm.!default {
	type plug
	slave.pcm rate48000Hz
}

pcm.rate48000Hz {
	type rate
	slave {
		pcm writeFile # Direct to the plugin which will write to a file
		format S16_LE
		rate 48000
		channels 2
	}
}

pcm.writeFile {
	type file
	slave.pcm null
	file "/tmp/snapfifo"
	format "raw"
}

**Comment by BugHunterPhilosopher** [comment]
Good afternoon,

I've managed to have it all work, at long last! I moved all my sound & video tasks on a NUC running [Audio Linux](http://www.audio-linux.com) and the official documentation from Snapcast has been enough to get me going. I still use the same 5.1 sound card, but I can not tell the difference in set up that can explain.

Anyway, thank you very much, I now have my own Open Source multi-room system.
